Portal/src/Datahub.Portal/Pages/Workspace/Toolbox/WorkspaceToolboxPage.razor

```diff
@@ -168,59 +187,200 @@
             _snackbar.Add(Localizer["{0} successfully added to your workspace", GetLabel(tool)], Severity.Success);
             await InvokeAsync(StateHasChanged);
         }
         catch (Exception e)
         {
             _logger.LogError(e, "Error adding tool {Tool} to workspace {WorkspaceAcronym}", tool, WorkspaceAcronym);
-            _toolCurrentDevelopmentStatusMap[tool] = ToolStatus.Error;
+            _toolCurrentDevelopmentStatusMap[tool] = ProjectResourceStatus.Error;
             _snackbar.Add(Localizer["Error adding {0} to your workspace", GetLabel(tool)], Severity.Error);
         }
     }
 
+    private async Task RemoveToolFromWorkspace(string tool)
+    {
+        _logger.LogInformation("Removing {Tool} from {WorkspaceAcronym}", GetLabel(tool), WorkspaceAcronym);
+        _toolCurrentDevelopmentStatusMap[tool] = ProjectResourceStatus.SendingDeletion;
+        await InvokeAsync(StateHasChanged);
+        try
+        {
+
+            var currentUser = await _userInformationService.GetCurrentPortalUserAsync();
+
+            await using var ctx = await _contextFactory.CreateDbContextAsync();
+            var _resourceCheck = await ctx.Project_Resources2
+                .FirstOrDefaultAsync(r => r.ResourceType.EndsWith(tool));
+
+
+            // adding deletion request to queue
+            var userEmail = await _userInformationService.GetUserEmail();
+            var workspaceDefinition = await _resourceMessagingService.GetWorkspaceDefinition(WorkspaceAcronym, requestingUserEmail: userEmail);
+            //await _resourceMessagingService.SendToTerraformDeleteResourceQueue(workspaceDefinition, _workspace.Project_ID, tool);
+
+            _logger.LogInformation("Tool {Tool} successfully removed from workspace {WorkspaceAcronym}", tool, WorkspaceAcronym);
+            _toolCurrentDevelopmentStatusMap[tool] = ProjectResourceStatus.Available;
+            _snackbar.Add(Localizer["{0} successfully removed from your workspace", GetLabel(tool)], Severity.Success);
+
+            _resourceCheck.Status = ProjectResourceStatus.Available.ToString();
+
+            await HandleSendEmail(tool);
+
+            await ctx.SaveChangesAsync();
+
+            await InvokeAsync(StateHasChanged);
+        }
+        catch (Exception e)
+        {
+            _logger.LogError(e, "Error removing tool {Tool} from workspace {WorkspaceAcronym}", tool, WorkspaceAcronym);
+            _toolCurrentDevelopmentStatusMap[tool] = ProjectResourceStatus.Error;
+            _snackbar.Add(Localizer["Error removing {0} from your workspace", GetLabel(tool)], Severity.Error);
+        }
+    }
+
+    private async void GoToResourceDeletionPage(string tool)
+    {
+        await using var ctx = await _contextFactory.CreateDbContextAsync();
+        var _resourceCheck = await ctx.Project_Resources2
+            .FirstOrDefaultAsync(r => r.ResourceType.EndsWith(tool));
+        _resourceCheck.Status = ProjectResourceStatus.PendingDeletion.ToString();
+
```

Comment:
You should use `TerraformTemplate.GetTerraformServiceType(string templateName)` to get the value that is used in the `ResourceType` column of the `Project_Resources`, instead of comparing the ends of the strings.
E.g:
```csharp
var resourceType = TerraformTemplate.GetTerraformServiceType(tool);
using var ctx = await _contextFactory.CreateDbContextAsync();
var _resourceCheck = await ctx.Project_Resources2.FirstOrDefaultAsync(r => r.ResourceType == resourceType);
```